File: 프로그래머스/2/42578. 의상/의상.java
import java.util.Map;
import java.util.HashMap;

class Solution {
    public int solution(String[][] clothes) {
        Map<String, Integer> map = new HashMap<>();
        
        for(String[] clo : clothes){
            String kind = clo[1];
            map.put(kind, map.getOrDefault(kind, 0) + 1);
        }
        
        int total = 1;
        for(Map.Entry<String, Integer> entry : map.entrySet()){
            total *= (entry.getValue() + 1);
        }
        
        return total - 1;
    }
}

File: 프로그래머스/2/42578. 의상/의상.kt
class Solution {
    fun solution(clothes: Array<Array<String>>): Int {
        var answer = 0
        val kindCount = HashMap<String, Int>()
        for((_, kind) in clothes){
            kindCount[kind] = kindCount.getOrDefault(kind, 0) +1
        }
        // 조합 공식으로 풀기
        var result = 1
        for((key, value) in kindCount.entries){
            //println("key : $key, value : $value")
            result *=(value+1)
        }
        return result-1
    }
}